File: data/readable_format/fortran/file958.f90
program test_qinit
    implicit none
    double precision, allocatable :: q(:,:)
    double precision, allocatable :: aux(:,:)
    integer :: mx, meqn, mbc, maxmx, maux
    double precision :: xlower, dx
    integer :: i

    ! Test parameters
    maxmx = 100
    meqn = 2
    mbc = 1
    mx = 5
    xlower = 0.0
    dx = 0.01
    maux = 0

    ! Allocate arrays
    allocate(q(1-mbc:maxmx+mbc,meqn))
    allocate(aux(1-mbc:maxmx+mbc,maux))

    ! Call the subroutine
    call qinit(maxmx,meqn,mbc,mx,xlower,dx,q,maux,aux)

    ! Check and print first few values of q for testing
    do i=1,5
        print *, 'q(', i, ',1) = ', q(i,1)
        print *, 'q(', i, ',2) = ', q(i,2)
    end do

    ! Deallocate arrays
    deallocate(q)
    deallocate(aux)
end program test_qinit

subroutine qinit(maxmx,meqn,mbc,mx,xlower,dx,q,maux,aux)
    implicit double precision (a-h,o-z)
    dimension q(1-mbc:maxmx+mbc, meqn)
    dimension aux(1-mbc:maxmx+mbc, *)
    common /cqinit/ beta
    integer i
    double precision xcell

    do 150 i=1,mx
        xcell = xlower + (i-0.5d0)*dx
        q(i,1) = 0.5d0*dexp(-80.d0*xcell**2)
        if (dabs(xcell+0.2d0) .lt. 0.1d0) then
            q(i,1) = q(i,1) + 0.5d0
        endif
        q(i,2) = 0.d0
150 continue
    return
end

File: data/readable_format/cpp/file958.cpp
#include <gtest/gtest.h>
#include <vector>
#include <cmath> // for exp() and abs()

// Assuming global variable (from common /cqinit/ in Fortran)
double beta; 

// The qinit function translated from Fortran to C++
void qinit(int maxmx, int meqn, int mbc, int mx, double xlower, double dx, std::vector<std::vector<double>>& q, int maux, std::vector<std::vector<double>>& aux) {
    for (int i = 0; i < mx; ++i) {
        double xcell = xlower + (i + 0.5) * dx;
        q[i][0] = 0.5 * exp(-80.0 * xcell * xcell);
        if (std::abs(xcell + 0.2) < 0.1) {
            q[i][0] += 0.5;
        }
        q[i][1] = 0.0;
    }
}

// Unit test for qinit function
TEST(QinitTest, InitializesCorrectly) {
    int maxmx = 100, meqn = 2, mbc = 1, mx = 5;
    double xlower = 0.0, dx = 0.01;
    std::vector<std::vector<double>> q(mx, std::vector<double>(meqn, 0.0));
    int maux = 0; // Example value, adjust as needed
    std::vector<std::vector<double>> aux; // Assuming aux isn't used in this simplified example

    qinit(maxmx, meqn, mbc, mx, xlower, dx, q, maux, aux);

    // Check the first few values of q
    EXPECT_NEAR(0.5 * exp(-80.0 * (0.005 * 0.005)), q[0][0], 1e-5);
    EXPECT_DOUBLE_EQ(0.0, q[0][1]);
    // Add more checks as necessary
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}